src/renderer/contexts/BackendContext.tsx

```diff
@@ -29,6 +36,8 @@ interface BackendContextState {
     scope: Scope;
     restartingRef: Readonly<React.MutableRefObject<boolean>>;
     restart: () => Promise<void>;
+    nodesQuery: UseQueryResult<BackendNodesResponse, unknown>;
+    nodesInfo: NodesInfo | undefined;
```

Comment:
No. `main.tsx` basically uses these 2 to deduce the current loading state. This is a suuuuper leaky abstraction, or rather no abstraction at all. So let's fix that. Maybe something like `connectionState: "connecting" | "connected" | "failed"` could work.